utils/common/src/main/java/org/apache/brooklyn/util/guava/Maybe.java

```diff
@@ -457,6 +457,8 @@ public boolean equals(Object obj) {
         Maybe<?> other = (Maybe<?>)obj;
         if (!isPresent()) 
             return !other.isPresent();
+        if (!other.isPresent())
+            return false;
```

Comment:
that seems harsh to me; could cause problems if we serialized a `Maybe`?

could compare the exceptions but for now I will just add javadoc